### PR TITLE
Update documentation for transitive dependency scanning

### PR DESCRIPTION
**Pull request by cuixq**

### CODE REVIEW

File: docs/supported_languages_and_lockfiles.md [hunk 1]
```diff
@@ -70,6 +70,14 @@ Submoduled dependencies are included in the project folder retain their Git hist
 
 Vendored dependencies have been directly copied into the project folder, but do not retain their Git histories. OSV-Scanner uses OSV's [determineversion API](https://google.github.io/osv.dev/post-v1-determineversion/) to estimate each dependency's version (and associated Git Commit). Vulnerabilities for the estimated version are returned. This process requires no additional work from the user. Run OSV-Scanner as you normally would.
 
+## Transitive dependency scanning
+
+OSV-Scanner now supports transitive dependency scanning for Maven pom.xml. This feature is enabled by default when scanning, but it is disabled in the [experimental offline mode](./offline-mode.md).
```
Comment (oliverchang) [inline docs/supported_languages_and_lockfiles.md:75]: nit: "now" is not required for documentation. it makes more sense in the context of an announcement.

File: docs/supported_languages_and_lockfiles.md [hunk 2]
```diff
@@ -70,6 +70,14 @@ Submoduled dependencies are included in the project folder retain their Git hist
...
+
+OSV-Scanner uses [deps.dev’s resolver library](https://github.com/google/deps.dev/tree/main/util/resolve) to compute the dependency graph of a project. This graph includes all of the direct and transitive dependencies. Requirements are provided by [deps.dev API](https://docs.deps.dev/api/v3/#getrequirements).
```
Comment (oliverchang) [inline docs/supported_languages_and_lockfiles.md:77]: Should we mention that this also uses deps.dev's database/API by default for package versions? And that support is coming soon for private registries (link to issue).

**Review by oliverchang — COMMENTED** [review]
LG with some minor comments.

File: docs/supported_languages_and_lockfiles.md (hunk 2 above)
Comment (cuixq) [reply docs/supported_languages_and_lockfiles.md:77]: just made a new issue about private registry support https://github.com/google/osv-scanner/issues/1045 and link it in the documentation

File: docs/supported_languages_and_lockfiles.md [hunk 3]
```diff
@@ -70,6 +70,14 @@ Submoduled dependencies are included in the project folder retain their Git hist
 
 Vendored dependencies have been directly copied into the project folder, but do not retain their Git histories. OSV-Scanner uses OSV's [determineversion API](https://google.github.io/osv.dev/post-v1-determineversion/) to estimate each dependency's version (and associated Git Commit). Vulnerabilities for the estimated version are returned. This process requires no additional work from the user. Run OSV-Scanner as you normally would.
 
+## Transitive dependency scanning
+
+OSV-Scanner supports transitive dependency scanning for Maven pom.xml. This feature is enabled by default when scanning, but it is disabled in the [experimental offline mode](./offline-mode.md).
```
Comment (another-rex) [inline docs/supported_languages_and_lockfiles.md:75]:
```suggestion
OSV-Scanner supports transitive dependency scanning for Maven pom.xml. This feature is enabled by default when scanning, but it is disabled in the [offline mode](./offline-mode.md).
```

I don't think we need to mention that it's experimental here, as it's mentioned in the offline-mode document.

File: docs/supported_languages_and_lockfiles.md [hunk 4]
```diff
@@ -70,6 +70,14 @@ Submoduled dependencies are included in the project folder retain their Git hist
...
+
+OSV-Scanner uses [deps.dev’s resolver library](https://pkg.go.dev/deps.dev/util/resolve) to compute the dependency graph of a project. This graph includes all of the direct and transitive dependencies. By default, [deps.dev API](https://docs.deps.dev/api/v3/index.html) is queried for package versions and requirements. The support for private registries is [coming soon](https://github.com/google/osv-scanner/issues/1045).
+
+After the dependency resolution, the OSV database is queried for the vulnerabilities associated with these dependencies.
```
Comment (another-rex) [inline docs/supported_languages_and_lockfiles.md:79]:
```suggestion
After the dependency resolution, the OSV database is queried for the vulnerabilities associated with these dependencies as usual.
```

**Review by another-rex — APPROVED** [review]
LGTM, some minor suggestions

**Review by hogo6002 — APPROVED** [review]

**Pull request merged**